overlord/ifacestate/helpers.go

```diff
@@ -164,6 +166,9 @@ func (m *InterfaceManager) regenerateAllSecurityProfiles() error {
 		}
 	}
 
+	writeSystemKey := true
+	os.Remove(dirs.SnapSystemKeyFile)
```

Comment:
this means snap-confine that might run in parallel will wait for the key to appear again, right?

Reply:
Yes. Ar the same time perhaps this is exactly what should happen given that system is partially inconsistent during this operation.